File: go/problem219_contains_duplicate_ii.go
// https://leetcode.com/problems/contains-duplicate-ii 

func containsNearbyDuplicate(nums []int, k int) bool {
    seen := make(map[int]int)

    for i, num := range nums {
        index, exist := seen[num]
        if exist && i-index<=k {
            return true
        } 
        seen[num] = i
    }

    return false
}

File: java/Problem219_ContainsDuplicateII.java
// https://leetcode.com/problems/contains-duplicate-ii/

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer,Integer> map = new HashMap<>();

        for (int i=0; i<nums.length; i++) {
            int value = nums[i];
            if (map.containsKey(value)) {
                int dif = i - map.get(value);

                if (dif <= k) return true;
            }
            map.put(value, i);
        }

        return false;
    }
}
